sigstore/verify/models.py

```diff
@@ -244,11 +246,30 @@ def from_bundle(
 
         if len(certs) == 0:
             raise InvalidMaterials("expected non-empty certificate chain in bundle")
-        cert_pem = PEMCert(
-            load_der_x509_certificate(certs[0].raw_bytes)
-            .public_bytes(Encoding.PEM)
-            .decode()
-        )
+
+        # Per client policy in protobuf-specs: the first entry in the chain
+        # MUST be a leaf certificate, and the rest of the chain MUST NOT
+        # include a root CA or any intermediate CAs that appear in an
+        # independent root of trust.
+        #
+        # We expect some old bundles to violate the rules around root
+        # and intermediate CAs, so we issue warnings and not hard errors
+        # in those cases.
+        leaf_raw, *chain_raw = certs
```

Comment:
Nit: Maybe do the certificate loading ahead of time? Then the code below will only contain the checks rather than doing it in two different spots.
```python
leaf_certs, *chain_raw = [load_der_x509_certificate(cert.raw_bytes) for cert in certs]

# Now do checks...
```

Reply:
Good call!

Reply:
[728f7f2](https://github.com/sigstore/sigstore-python/pull/632/commits/728f7f2c92b4c7b2d990f004163b28d45d07a6c2)